tests/run-coverage/lazy_boolean.coverage

```diff
@@ -32,7 +32,7 @@
                                           ^0
    LL|       |
    LL|       |    if
-   LL|      1|        !
+   LL|       |        !
    LL|      1|        is_true
    LL|      0|    {
```

Comment:
From a coverage perspective this seems wrong; the `!` operator should have the same execution count as its operand.

However, at a glance I can't say whether this PR is actually doing something wrong, or just exposing a latent issue in the coverage instrumentor's heuristics.

(I'm not particularly concerned about this being a problem in practice; I just want to explicitly note it as something for me to keep my eye on.)

Reply:
It might be the case that because `!` in `if` condition is now directly lowered into `switchInt`s, by-passing the operator evaluation, the profiler might not pick it up. In other contexts, such as boolean expressions, `!` should still show up I think.

Reply:
That makes sense. So `if !cond { a } else { b }` gets lowered as though it were `if cond { b } else { a }`. The `!` doesn’t exist in MIR, and the coverage pass doesn’t have a special case to detect this, so it isn’t included in the coverage region.

This seems acceptable for now. The mappings are slightly worse, but they aren’t *lying*, and most coverage users probably won’t notice the difference.